### PR TITLE
Remove cylc submit and cylc jobscript

### PR DESCRIPTION
**Pull request by datamel**
This PR also involves rewriting some functional tests as unit tests.

Note, I have not removed all of the functional tests in the `jobscript` directory. I believe the ones I have left are useful functional tests that test jobscript and are more suited to functional tests than unit tests.

The functional tests removed have been replaced by unit tests, with the exception of   `tests/jobscript/01-multi.t` - this currently relies on `cylc jobscript` command and is not suited to unit testing. This is testing that `namespace_hierarchy` is correctly populated and is covered by other tests.

These changes relate to #1249

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Appropriate tests are included (unit and/or functional).
- [x] Appropriate change log entry included.
- [x] I will open a documentation PR at cylc/cylc-doc

### CODE REVIEW

File: cylc/flow/scripts/cylc.py [hunk 1]
```diff
@@ -24,7 +24,7 @@
 from cylc.flow import __version__
 
 # These will be ported to python as click commands
-bash_commands = ["cylc-graph-diff", "cylc-jobscript"]
+bash_commands = ["cylc-graph-diff"]
```
Comment (oliver-sanders) [inline cylc/flow/scripts/cylc.py:27]: Just one left!

File: cylc/flow/tests/test_job_file.py [hunk 2]
```diff
@@ -30,28 +33,428 @@
...
-            with TemporaryFile(mode="w+") as handle:
-                JobFileWriter()._write_prelude(handle, job_conf)
-        self.assertIn("bad cylc executable", str(ex.exception))
+        mocked_get_remote_suite_run_dir.return_value = "run/dir"
+        JobFileWriter().write(local_job_file_path, job_conf)
+
+        assert (os.path.exists(local_job_file_path))
+        size_of_file = os.stat(local_job_file_path).st_size
+        print(size_of_file)
+        assert(size_of_file == 1715)
```
Comment (oliver-sanders) [inline cylc/flow/tests/test_job_file.py:102]: That's a somewhat strange way to test it!

This is effectively a "known good output" test, so we could just include the "good output", that way when the test breaks you get a diff.

File: cylc/flow/scripts/cylc.py (hunk 1 above)
Comment (datamel) [reply cylc/flow/scripts/cylc.py:27]: Wooty! Are there plans for this one to go too?

Comment (oliver-sanders) [reply cylc/flow/scripts/cylc.py:27]: We might be able to merge it into `cylc graph` e.g. `cylc graph --diff`.

**Comment by oliver-sanders** [comment]
> Later I think we also need to update the cylc-docs?

The CLI docs are probably a bit out-of-wack, once we've got the CLI properly implemented we will auto-document the CLI.

**Comment by hjoliver** [comment]
@kinow 's links above do point at references to `cylc job-submit` in the User Guide text, so it's not just the "CLI docs" (if by that you mean the the auto-generated command help docs).

**Comment by oliver-sanders** [comment]
True, though there is something we can do to automate this.

We now write cylc configurations in this syntax:

```rst
You can change this with :cylc:conf:`suite.rc[scheduling]max active cycle points`.
```

If we renamed or removed `max active cycle points` then the docs would fail to build due to a broken reference. We can do the same thing with commands:

```rst
You can see what the job script looks like using the :command:`cylc jobscript` command.
```

**Pull request merged**